### ISSUE TITLE
CollapsiblePanel option

### ISSUE THREAD

**Issue by veragreen119** [syonfox/leaflet-panel-layers#3]
Add  collapsiblePanel option  - same as collapsibleGroups but acts one level up on the panel object.

- collapsiblePanel  - allow the entire panel to be collapsed.  

- If fully collapsed with the  "collapsed: true" mode ("B" in screenshot) only the Panel Icon will show.  You might be able to use the leaflet default as the default if that makes sense.

- If fully collapsed with the  "collapsed: false" mode ("A" in screenshot)  it would show the icon and then the panel title. 

![image](https://user-images.githubusercontent.com/40037166/57742567-d69bc380-767e-11e9-96c1-3e8e2f2c148d.png)

**Comment by syonfox** [comment]
I made a basic version of this we need to talk about style changes a bit one idea is to have an icon instead of the +/- and possibly make it turn Grey or red when collapsed

![image](https://user-images.githubusercontent.com/9345677/57811492-38ac0580-771f-11e9-993f-6193fc5eb217.png)

![image](https://user-images.githubusercontent.com/9345677/57811525-524d4d00-771f-11e9-8f91-c9d012d5997d.png)

![image](https://user-images.githubusercontent.com/9345677/57811584-7446cf80-771f-11e9-94fe-1b1c4866b5e5.png)

**Comment by veragreen119** [comment]
Yeah that's a good idea.  The long term solution should allow the user to assign their own custom icon to the panel when it is fully collapsed to facilitate having multiple panels but to start with we can just assign a default icon. 

I am wondering if integration with the easy button plugin might be a good path forward? Not sure if that makes it harder or easier: [https://github.com/CliffCloud/Leaflet.EasyButton](url)

If it will work with glyphicon we could use these:
glyphicon glyphicon-plus
glyphicon glyphicon-minus
OR
glyphicon glyphicon-chevron-up
glyphicon glyphicon-chevron-down
[https://getbootstrap.com/docs/3.3/components/](url)

Or we can use downloaded icons, the simple up and down arrow in this set isn't bad:
[https://freeicons.io/icon-list/technology-icons](url)
I uploaded these to examples/images/icons.  Not sure how to delete files but you can delete the icons that are named with a long number.  I uploaded them again with a useful name.

**Comment by veragreen119** [comment]
The easy button method works.  So if we required easyButton as a dependency for this plugin maybe you can use similar code within leaflet-panel-layers.js? This is where my programming background runs thin so let me know if this is a bad approach . . .  

The other approach is to scrap this enhancement and just tell people in documentation they can use easyButton if they want the menu to fully collapse into an icon . . . 

```
                var ctlOver1Button = L.easyButton({
                    states: [{
                            stateName: 'open-over1',        // name the state
                            icon:      'glyphicon-chevron-down',               // and define its properties
                            title:     'close over1',      // like its title
                            onClick: function(btn) {       // and its callback
                                map.addControl(over1);
                                btn.state('close-over1');    // change state on click!
                            }
                        }, {
                            stateName: 'close-over1',
                            icon:      'glyphicon-chevron-up',
                            title:     'open over1',
                            onClick: function(btn) {
                                over1.remove();
                                btn.state('open-over1');
                            }
                    }],
                    position:'topleft'
                }).addTo(map);
```

**Comment by veragreen119** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Test](https://trello.com/c/QpRSGmS9/8-test)

**Comment by syonfox** [comment]
is implemented as of b8a79a5c04e6d88edd8833d4eb9bc169c917bf6c if there are bugs or improvements please open an new issue or reopen this one
